db/db_universal_compaction_test.cc

```diff
@@ -2230,6 +2231,209 @@ TEST_F(DBTestUniversalCompaction2, PeriodicCompaction) {
   ASSERT_EQ(4, output_level);
 }
 
+TEST_F(DBTestUniversalCompaction2, PeriodicCompactionOffpeak) {
```

Comment:
Is there a significant difference in universal and leveled behavior here? It seems like a lot of duplication (more or less).

Reply:
IMHO, code duplication in tests (more or less) is a greater sin than keeping tests neatly organized by feature. Features interact, so there's no way to be absolutely clean in organization--assuming your tests are interesting and not just a rehash of the implementation; you just put things some place reasonable and respect the greater threats to maintenance burden such as verbosity / unnecessary repetition.

It would probably be different if compaction strategies were pluggable components, but they are deeply integrated.

I appreciate the narrative comments to help future engineers to understand what is being tested and how. Clarity in the "what & why" of a test is most important. :) (Sometimes it's obvious: feature X has its documented effect when turned on and does not when it's turned off.)

Reply:
Simplified the test for Leveled one, added comment and addressed the feedback in the Universal one.